pkg/v1/tkg/client/validate.go

```diff
@@ -1790,6 +1790,20 @@ func (c *TkgClient) validateNameservers(nameserverConfigVariable string) error {
 		return nil
 	}
 
+	featureActivated := false
+	switch clusterRole {
+	case TkgLabelClusterRoleManagement:
+		featureActivated = c.IsFeatureActivated(config.FeatureFlagManagementClusterCustomNameservers)
+	case TkgLabelClusterRoleWorkload:
+		featureActivated = c.IsFeatureActivated(config.FeatureFlagClusterCustomNameservers)
+	}
+
+	if !featureActivated {
+		return fmt.Errorf("option %s is set to %q, but custom nameserver support is not enabled (because it is not fully functional). To enable custom nameservers, set features.management-cluster.custom-nameservers to \"true\"",
```

Comment:
I suggest setting the error message in the case statement, because the current error message only applies to management name clusters, but it might be a workload case.
Also (a nit) I would suggest making the message more explicit, something like:
`To enable custom nameservers, run the command: tanzu config set features.management-cluster.custom-nameservers true`

Reply:
Ah, yes, of course. That would be much better. Thanks!

Reply:
Fixed and pushed!